File: benchmark/csharp/Services/CryptoService.cs
using System.Security.Cryptography;
using System.Text;

namespace Benchmark.CSharp.Services
{
    public class CryptoService
    {
        public string HashData(string input)
        {
            // Broken Crypto Algorithm [CWE-327]
            using (MD5 md5 = MD5.Create())
            {
                byte[] inputBytes = Encoding.ASCII.GetBytes(input);
                byte[] hashBytes = md5.ComputeHash(inputBytes);
                return Encoding.ASCII.GetString(hashBytes);
            }
        }

        public void EncryptToken()
        {
            // Hardcoded Key [CWE-321]
            string key = "S3cr3t_K3y_1234567890";
            // Use key for encryption...
        }
    }
}

File: benchmark/scala/src/main/scala/services/CryptoService.scala
package services
import java.security.MessageDigest

object CryptoService {
  def hashData(data: String): String = {
    // Broken Crypto Algorithm [CWE-327]
    val md5 = MessageDigest.getInstance("MD5")
    md5.digest(data.getBytes).map("%02x".format(_)).mkString
  }

  def encryptData(): Unit = {
    // Hardcoded Key [CWE-321]
    val key = "S3cr3t_K3y_1234567890"
    println(s"Using key: $key")
  }
}
